source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOAdaptiveCard.mm

```diff
@@ -16,6 +17,41 @@
 @implementation ACOAdaptiveCard
 {
     std::shared_ptr<AdaptiveCard> _adaptiveCard;
+    NSMutableArray *_inputs;
+}
+
+- (void) setInputs:(NSMutableArray *) inputs
```

Comment:
>NSMutableArray * [](start = 20, length = 16)

You should take an NSArray* here and copy its contents. You should not share ownership of the array with the caller. If the caller modifies the array after this call completes, that will change your array as well. Similarly, and even worse, if you change the array (by having someone call appendInputs), that will change the caller's local copy of the array too.

Reply:
thanks!

---
In reply to: [172390193](https://github.com/Microsoft/AdaptiveCards/pull/1211#discussion_r172390193) [](ancestors = 172390193)

Reply:
that's right. thanks


---
In reply to: [172582709](https://github.com/Microsoft/AdaptiveCards/pull/1211#discussion_r172582709) [](ancestors = 172582709,172392306,172390193)